node/core/pvf/common/src/worker/security.rs

```diff
@@ -186,3 +186,98 @@ pub mod landlock {
 		}
 	}
 }
+
+// TODO: Add a check for whether seccomp is supported and warn if not, like we do for landlock.
+//       Ideally we would have <https://github.com/rust-vmm/seccompiler/issues/51>.
+//       In lieu of that we can try restricting a dummy thread like we do for landlock.
+/// Process sandboxing using seccomp, a Linux facility for blocking syscall execution.
+///
+/// Currently we only block two syscalls, `socket` and `socketpair`, to effectively disable
+/// networking.
+///
+/// # Action on Caught Syscall
+///
+/// TODO: Update with the action and explain why given action was chosen.
+#[cfg(target_os = "linux")]
+pub mod seccomp {
+	use seccompiler::*;
+	use std::collections::BTreeMap;
+
+	/// The action to take on caught syscalls.
+	#[cfg(not(test))]
+	const CAUGHT_ACTION: SeccompAction = SeccompAction::KillProcess;
+	/// Don't kill the process when testing...
+	#[cfg(test)]
+	const CAUGHT_ACTION: SeccompAction = SeccompAction::Errno(libc::EACCES as u32);
+
+	#[derive(thiserror::Error, Debug)]
+	pub enum Error {
+		#[error(transparent)]
+		Seccomp(#[from] seccompiler::Error),
+
+		#[error(transparent)]
+		Backend(#[from] seccompiler::BackendError),
+	}
+
+	pub type Result<T> = std::result::Result<T, Error>;
+
+	// TODO: Compile the filter at build-time rather than runtime.
```

Comment:
Yep. The simplest way to do it would probably be to generate the `BpfProgram` in `build.rs`, cast it to a `&[u8]` with `slice::from_raw_parts`, write it to file, and then `include_bytes!` it here. You could then `slice::from_raw_parts` it again into `&'a [sock_filter]` and pass it to `seccompiler::apply_filter`. (`sock_filter` is `#[repr(C)]` so this is safe to do)

Reply:
Yep, that's close to the instructions here: https://github.com/rust-vmm/seccompiler#compiling-filters.